lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -143,7 +143,7 @@ pthread_scheduler_uninit ()
   for (i = 0; i < scheduler.num_threads; ++i)
     {
       pthread_join (scheduler.thread_pool[i].thread, NULL);
-      PTHREAD_DESTROY_LOCK (&scheduler.thread_pool[i].lock);
+      // PTHREAD_DESTROY_LOCK (&scheduler.thread_pool[i].lock);
```

Comment:
This looks suspicious. Why? Otherwise looks OK to me.

Reply:
Isn't this a duplicate of https://github.com/pocl/pocl/blob/d899c94eae97c2a33ad8c38e96b0fb7d7e6d5893/lib/CL/devices/pthread/pthread_scheduler.c#L579 ?

Reply:
@isuruf yeah it is a duplicate. But anyway there's more to fix here. Can you drop the locking commit  from the PR please, i'll take care of the locks. Thanks